Configure

```diff
@@ -957,6 +960,11 @@ while (@argvcopy)
                         {
                         delete $disabled{"zstd"};
                         }
+                elsif ($1 eq "fips-jitter")
+                        {
+                        delete $disabled{"fips"};
+                        delete $disabled{"jitter"};
```

Comment:
Is this right?
Could you potentially want to build fips provider with jitter but not use it in the default provider?

Reply:
This just means that it is also available in the FIPS provider.  It won't be used unless config specifies it or the appropriate call is made to enable it.

I.e. this doesn't automatically make it _used_ in the default provider.

Reply:
Looking at build.info.. It does this

````
SUBDIRS=seeding

$RANDS_GOAL=../../libdefault.a ../../libfips.a

SOURCE[$RANDS_GOAL]=drbg.c test_rng.c drbg_ctr.c drbg_hash.c drbg_hmac.c
SOURCE[../../libdefault.a]=seed_src.c seed_src_jitter.c
SOURCE[../../libfips.a]=fips_crng_test.c
````

So the seed_src_jitter.c file is doing the check (which it probably shouldnt)..

The base and default provider both do this
````
static const OSSL_ALGORITHM deflt_rands[] = {
#ifndef OPENSSL_NO_JITTER
    { PROV_NAMES_JITTER, "provider=default", ossl_jitter_functions },
#endif
````

So enabling fips-jitter adds code to non fips source, which is what I disagree with.

Reply:
Hmmmm.....  the other  jitter options also have to be specified to locate the jitter library and include directory and they don't have fips in the name.  Maybe I should document that fips-jitter also enables jitter or insist that both configure options are specified?  Having the code in the default provider is harmless since it won't be used but introduces a small amount of bloat.

Dissociating the two will be painful I suspect.

Reply:
possibly a non-issue, as most do not perform a single configure and build of default and fips providers. It is quite typical to configure one tree and install headers, libcrypto, libssl from it; and another one to just get fips.so from. meaning most often default provider / libcrypto is discarded in the fips builds. But having this cleaner would be nice. Because if one only intents to use fips.so, one really does not need jitter in libcrypto.

Separately, I have been pondering a no-default build..... which doesn't link libdefault.a into libcrypto, such that default libcrypto only has base provider. Specifically for builds that only care about fips.so and nothing else. As it makes it possible to provide libcrypto for fips-certified-only environments witout ability to programatically load and use '-fips/fips=no' algorithms.

Reply:
Good point.  Once the first CVE fix comes out, split builds will be normal I suspect.